src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -121,9 +126,31 @@ private static bool IsEndOfLine(SyntaxTrivia trivia)
                     || trivia.IsDirective;
             }
 
-            private static bool HasEndOfLine(SyntaxTriviaList list)
+            /// <summary>
+            /// Returns the first end of line found in a <see cref="SyntaxTriviaList"/>.
+            /// </summary>
+            private static SyntaxTrivia? GetEndOfLine(SyntaxTriviaList list)
             {
-                return list.Any(t => IsEndOfLine(t));
+                foreach (var trivia in list)
+                {
+                    if (trivia.Kind() == SyntaxKind.EndOfLineTrivia)
+                    {
+                        return trivia;
+                    }
+
+                    if (trivia.IsDirective && trivia.GetStructure() is DirectiveTriviaSyntax directive)
+                    {
+                        foreach (var directiveTrivia in directive.EndOfDirectiveToken.TrailingTrivia)
```

Comment:
this inner foreach feels redundant.  could we not just call "GetEndOfLine" recursively on 'directive.EndOfDirectiveToken'?

Reply:
You are right, corrected.